addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/service/NetworkService.java

```diff
@@ -110,35 +112,77 @@ public void setUseSystemPing(boolean useSystemPing) {
-        Runnable runnable = new Runnable() {
+    /**
+     * Create a runner to be called by a scheduler periodically. The runner
+     * will update the device state and inform the callback listener.
+     *
+     * @return Return a runner.
+     */
+    private Runnable createUpdateListenerRunnable() {
+        return new Runnable() {
             @Override
             public void run() {
                 try {
-                    stateUpdate.newState(updateDeviceState());
+                    updateListener.updatedDeviceState(updateDeviceState());
                 } catch (InvalidConfigurationException e) {
-                    stateUpdate.invalidConfig();
+                    updateListener.invalidConfig();
                 }
             }
         };
+    }
 
-        refreshJob = scheduledExecutorService.scheduleAtFixedRate(runnable, 0, refreshInterval, TimeUnit.MILLISECONDS);
-
-        if (dhcplisten) {
-            try {
-                ReceiveDHCPRequestPackets.register(InetAddress.getByName(hostname).getHostAddress(), stateUpdate);
-            } catch (SocketException | UnknownHostException e) {
-                logger.error("Cannot use DHCP listen: {}", e.getMessage());
+    /**
+     * Cancel a running refreshJob. Will not throw if no job is running
+     * or the job is already canceled.
+     */
+    private void cancelRefreshJob() {
+        try {
+            if (refreshJob != null) {
+                refreshJob.cancel(true);
             }
+        } catch (Exception ignore) {
         }
     }
 
+    /**
+     * Start/Restart a fixed scheduled runner to update the devices reach-ability state.
+     *
+     * @param scheduledExecutorService A scheduler to run pings periodically.
+     * @param stateChangeListener Your callback listener
+     */
+    public void startAutomaticRefresh(ScheduledExecutorService scheduledExecutorService) {
+        cancelRefreshJob();
+        refreshJob = scheduledExecutorService.scheduleAtFixedRate(createUpdateListenerRunnable(), 0, refreshInterval,
+                TimeUnit.MILLISECONDS);
+
+        enableDHCPListen(dhcplisten);
+    }
+
     public void stopAutomaticRefresh() {
-        refreshJob.cancel(true);
+        cancelRefreshJob();
+        enableDHCPListen(false);
+    }
+
+    /**
+     * Registers the hostname of this network service object to the dhcp request packet listener if enabled.
+     * The hostname will be unregistered otherwise.
+     *
+     * @param enabled
+     */
+    private void enableDHCPListen(boolean enabled) {
         try {
-            ReceiveDHCPRequestPackets.unregister(InetAddress.getByName(hostname).getHostAddress());
+            if (enabled) {
+                try {
+                    ReceiveDHCPRequestPackets.register(InetAddress.getByName(hostname).getHostAddress(),
+                            updateListener);
+                } catch (SocketException e) {
+                    logger.error("Cannot use DHCP listen: {}", e.getMessage());
```

Comment:
Does this qualify as an error? (= fatal user mis-configuration or a bug in your code that must be reported?)